script/DeployTokenWithProxy.s.sol

```diff
@@ -11,13 +11,26 @@ contract DeployTokenWithProxy is BaseScript {
     }
 
     function deploy() public returns (ERC1967Proxy) {
+        // Read desired token cap from env or use default
+        uint256 defaultCap = vm.envOr({ name: "TOKEN_CAP", defaultValue: uint256(1_000_000 * 10 ** 18) });
+
         // Deploy the initial implementation
         address implementation = address(new TestStableToken());
 
-        // Encode the initialize call
-        bytes memory data = abi.encodeCall(TestStableToken.initialize, (1_000_000 * 10 ** 18));
+        // Encode the initialize call (cap)
+        bytes memory initData = abi.encodeCall(TestStableToken.initialize, (defaultCap));
 
         // Deploy the proxy with initialization data
-        return new ERC1967Proxy(implementation, data);
+        ERC1967Proxy proxy = new ERC1967Proxy(implementation, initData);
+
+        // Post-deploy assertions to ensure initialization succeeded
+        // These revert the script if validation fails.
+        address proxyAddr = address(proxy);
+
+        // Check cap set
+        uint256 actualMax = TestStableToken(proxyAddr).cap();
+        if (actualMax != defaultCap) revert("Proxy token cap mismatch after initialization");
```

Comment:
I which scenario can `actualMax` be unequal to `defaultCap` right after initialization?